int8 result
---
int64 left_encoder
int64 right_encoder
